bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigTomlProvider.java

```diff
@@ -86,12 +89,14 @@
  */
 public class ConfigTomlProvider implements ConfigProvider {
 
+    private final Set<String> requiredModules;
     Map<Module, TomlTableNode> moduleTomlNodeMap = new HashMap<>();
 
     TomlTableNode tomlNode;
 
-    public ConfigTomlProvider(Path configPath) {
-        this.tomlNode = getConfigurationData(configPath);
+    public ConfigTomlProvider(Path configPath, Map<Module, VariableKey[]> configVarMap) {
+        this.requiredModules = getRequiredModules(configVarMap);
```

Comment:
I think we do not need the required module check( Inside that we check the key). ConfigResolver will handle that with if (varKey.isRequired && configValue.isEmpty()) { check after getting config value